modules/telestion-application/src/main/java/de/wuespace/telestion/application/loader/Loader.java

```diff
@@ -0,0 +1,350 @@
+	 * {@link #onBeforeVerticleDeploy(Promise, Deployment) onBeforeVerticleDeploy} ➔
+	 * <strong><code>onAfterVerticleDeploy</code></strong>
+	 *
+	 * @param completePromise the promise that indicates the completion state of this loader
+	 * @param deployment      the deployment which contains the {@link io.vertx.core.Verticle Verticle} that deploys
+	 */
+	void onAfterVerticleDeploy(Promise<Void> completePromise, Deployment deployment) throws Exception;
+
+	/**
+	 * Gets called, before the {@link ConfigDeploymentLauncher Launcher} un-deploys a running
+	 * {@link io.vertx.core.Verticle Verticle}.
+	 * <p>
+	 * Use this event if you want to run steps before a verticle un-deploys.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution. A failed promise <strong>prevents</strong> the verticle from un-deploying.
+	 * <p>
+	 * The launcher calls this event handler. <strong>Don't call it by yourself.</strong>
+	 * <p>
+	 * <em>Call chain</em>:
+	 * <strong><code>onBeforeVerticleUndeploy</code></strong> ➔
+	 * {@link #onAfterVerticleUndeploy(Promise, Deployment) onAfterVerticleUndeploy}
+	 *
+	 * @param completePromise the promise that indicates the completion state of this loader
+	 * @param deployment      the deployment which contains the {@link io.vertx.core.Verticle Verticle} that un-deploys
+	 */
+	void onBeforeVerticleUndeploy(Promise<Void> completePromise, Deployment deployment) throws Exception;
+
+	/**
+	 * Gets called, after the {@link ConfigDeploymentLauncher Launcher} un-deploys a running
+	 * {@link io.vertx.core.Verticle Verticle}.
+	 * <p>
+	 * Use this event if you want to run steps after a verticle un-deploys.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution. A failed promise <strong>prevents</strong> the verticle from un-deploying.
+	 * <p>
+	 * The launcher calls this event handler. <strong>Don't call it by yourself.</strong>
+	 * <p>
+	 * <em>Call chain</em>:
+	 * {@link #onBeforeVerticleUndeploy(Promise, Deployment) onBeforeVerticleUndeploy} ➔
+	 * <strong><code>onAfterVerticleUndeploy</code></strong>
+	 *
+	 * @param completePromise the promise that indicates the completion state of this loader
+	 * @param deployment      the deployment which contains the {@link io.vertx.core.Verticle Verticle} that un-deploys
+	 */
+	void onAfterVerticleUndeploy(Promise<Void> completePromise, Deployment deployment) throws Exception;
+
+	/**
+	 * Gets called, when the running {@link io.vertx.core.Vertx Vertx} instance in the
+	 * {@link ConfigDeploymentLauncher Launcher} encounters an uncaught exception.
+	 * <p>
+	 * Use this event if you want to run steps on uncaught exception on the Vertx instance.
+	 * <p>
+	 * The passed promise should resolve when this loader finishes its execution and fail if the loader encounter
+	 * an error during execution.
```

Comment:
So ... what happens then? Do I get into an infitinite loop if my error handler keeps throwing errors? 😁

Reply:
Done in [`46e0777` (#501)](https://github.com/wuespace/telestion-core/pull/501/commits/46e07771cc2222f313d84472f69aa37a80b05033). Please re-review if you don't like it. :wink: